docs/backend_guide.md

```diff
@@ -0,0 +1,227 @@
+        * The Power Pay App, which allows users to create accounts, access transaction history, and transfer money, can also be developed using React JS to deliver a rich and responsive user experience.
+
+    7. Postman
+
+        Postman will be utilized as the testing tool for the API. As a widely adopted API testing tool, Postman offers a user-friendly interface that simplifies the process of sending HTTP requests and receiving responses. It empowers developers to customize various aspects of the API requests, such as headers, query parameters, request bodies, and authentication methods.
+
+        With Postman, we can efficiently organize our API requests into collections, facilitating the management and execution of multiple requests as part of a workflow or test suite. Additionally, we can create environments within Postman to store variables specific to different environments (e.g., development, staging, production), streamlining the management of environment-specific configurations.
+
+
+# API Documentation
+
+    The Power Payment Backend exposes various APIs with the following endpoints:
+
+        - POST /api/users: Create a new user account. Request body should include user details.
+        - POST /api/payments: Initiate a money transfer between users. Request body should include transfer details.
+        - GET /api/payments/{paymentId}: Get details of a specific payment by ID.
+        - GET /api/users/{userId}/transactions: Get transaction history for a specific user.
+        - DELETE /api/users/{userId}: Delete a user account by ID.
+
+For each endpoint, provide examples of requests and responses along with the expected request and response formats.
+
+
+# Database Schema
+
+    The Power Payment Backend is a secure system for seamless money transfers. Here's an example of the database schema that supports it.
+
+    - User Table:
+
+        user_id (Primary Key)
+        username
+        email
+        password
+        date_created
+
+    - Transaction History Table:
+
+        transaction_history_id (Primary Key)
+        transaction_id (Foreign Key referencing Transaction Table)
+        status (e.g., pending, completed, failed)
+        date_updated
+
+    - Transaction Table:
+
+        transaction_id (Primary Key)
+        sender_id (Foreign Key referencing User Table)
+        receiver_id (Foreign Key referencing User Table)
+        amount
+        date_created
+
+This schema includes tables for users, transactions, and transaction history. The user table stores user information, including username, email, and password. The transaction table tracks payment transactions between users, including sender, receiver, and amount. The transaction history table records the status and update timestamps for each transaction. Keep in mind that this is a basic example, and the actual database schema for the Power Pay App may require additional tables and relationships. 
+
+
+# Authentication
```

Comment:
Ok, this is the tricky part of our system. We want to allow users only from our domain. This means that we're not gonna allow some hackers coming from different websites. Also we'll be communicating with the translator service for exhanging SMS.

1. How can we ensure a Auth request is coming from our system?
2. How can I ensure the translator service is not compromised or fake?

This is what you should discuss here. The authentication seen by the user is cute but non-technical.

Reply:
To help you understand this, you know that our system doesn't use keycloak or similar oauth2 provider for security. Meaning we should implement security measures ourselves. We should then deeply understand how to secure the PWA by itself, when it interacts with our backend, without even talking about authentication. Same goes for the Translator service.